utils/src/main/kotlin/OkHttpClientHelper.kt

```diff
@@ -59,38 +75,22 @@ object OkHttpClientHelper {
      * Build a preconfigured client that uses a cache directory inside the [ORT data directory][ortDataDirectory].
      * Proxy environment variables are by default respected, but the client can further be configured via the [block].
      */
-    fun buildClient(block: OkHttpClient.Builder.() -> Unit = {}): OkHttpClient =
-        clients.getOrPut(block) {
-            val cacheDirectory = ortDataDirectory.resolve("cache/http")
-            val cache = Cache(cacheDirectory, MAX_CACHE_SIZE_IN_BYTES)
-            val specs = listOf(ConnectionSpec.MODERN_TLS, ConnectionSpec.COMPATIBLE_TLS, ConnectionSpec.CLEARTEXT)
-
-            // OkHttp uses Java's global ProxySelector by default, but the Authenticator for a proxy needs to be set
-            // explicitly. Also note that the (non-proxy) authenticator is set here and is primarily intended for
-            // "reactive" authentication, but most often "preemptive" authentication via headers is required. For proxy
-            // authentication, OkHttp emulates preemptive authentication by sending a fake "OkHttp-Preemptive" response
-            // to the reactive proxy authenticator.
-            OkHttpClient.Builder()
-                .cache(cache)
-                .connectionSpecs(specs)
-                .readTimeout(Duration.ofSeconds(30))
-                .authenticator(Authenticator.JAVA_NET_AUTHENTICATOR)
-                .proxyAuthenticator(Authenticator.JAVA_NET_AUTHENTICATOR)
-                .apply(block)
-                .build()
-        }
+    fun buildClient(block: (OkHttpClient.Builder.() -> Unit)? = null): OkHttpClient =
+        block?.let {
+            clients.getOrPut(it) { client.newBuilder().apply(block).build() }
+        } ?: client
 
     /**
      * Execute a [request] using the client for the specified [builder configuration][block].
      */
-    fun execute(request: Request, block: OkHttpClient.Builder.() -> Unit = {}): Response =
+    fun execute(request: Request, block: (OkHttpClient.Builder.() -> Unit)? = null): Response =
```

Comment:
While this function certainly simplifies some use cases, I am a bit afraid that it also fosters bad practice: I think if a component executes HTTP requests and needs a customized HTTP client for this purpose, it should explicitly obtain such a client via `buildClient()` and use it for all its requests. Otherwise, there is a danger that `execute()` is called with different initializer blocks, causing the creation of multiple HTTP clients for this component.

So maybe just limit `execute()` to the shared default client and removing the `block` argument?

Reply:
I'm not sure I get that danger you're referring to.  When you say

> Otherwise, there is a danger that `execute()` is called with different initializer blocks

why would a user call `execute()` with different initializer blocks if he / she does not really need different blocks?

Reply:
I meant a scenario as follows: A component, say an advisor or a (remote) scanner, uses the `execute()` function for convenience and has somewhere a code block like:

```
val request = ....
val response = OkHttpClientHelper.execute(request) { 
  // some customization
}

```
Now by applying the copy & paste pattern the statement to execute requests is copied multiple times in the code base. Then I think this would lead to the creation of multiple HTTP client instances. (Or would the compiler be smart enough to figure out that the initialization block is always the same?) Of course, the code quality of such a client is rather poor.

But viewed from the other side: What is the concrete use case for calling `execute()` with an initializer block? IIUC, a valid case could only be the execution of a single request ever that requires specific customization. Do we have such cases or is it rather typical to send multiple requests?

Reply:
> Or would the compiler be smart enough to figure out that the initialization block is always the same?

I don't think it's even a matter of the compiler optimizing something. In all the copied occurrences the second argument is the default of `null`, so the code expands to `client.newCall(request).execute()`, which always uses the same client instance, Or am I missing something?

> What is the concrete use case for calling `execute()` with an initializer block?

Well, a use-case would be a single-shot request that needs a different timeout value, for example, because a specific server is known to be slow. But indeed, we currently never call this function with a custom `block`; the feature is basically just there because it was trivial to add, and of potential use.

Reply:
Thanks for clarifying. I won't insist any further.